frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -607,37 +574,55 @@ export const sessionRecordingDataLogic = kea<sessionRecordingDataLogicType>([
                         return null
                     }
 
+                    const sessionEventsQuery = hogql`
+                            SELECT uuid, event, timestamp, elements_chain, properties.$window_id, properties.$current_url, properties.$event_type
+                            FROM events
+                            WHERE timestamp > ${start.subtract(BUFFER_MS, 'ms')}
+                              AND timestamp < ${end.add(BUFFER_MS, 'ms')}
+                              AND $session_id = ${props.sessionRecordingId}
+                              ORDER BY timestamp ASC
+                        LIMIT 1000000
+                        `
+
+                    let relatedEventsQuery = hogql`
+                            SELECT uuid, event, timestamp, elements_chain, properties.$window_id, properties.$current_url, properties.$event_type
+                            FROM events
+                            WHERE timestamp > ${start.subtract(BUFFER_MS, 'ms')}
+                              AND timestamp < ${end.add(BUFFER_MS, 'ms')}
+                              AND (empty($session_id) OR isNull($session_id)) AND properties.$lib != 'web'
+                        `
+                    if (person?.uuid) {
+                        relatedEventsQuery += `
+                            AND person_id = '${person.uuid}'
+                        `
+                    }
+                    if (!person?.uuid && values.sessionPlayerMetaData?.distinct_id) {
+                        relatedEventsQuery += `
+                            AND distinct_id = ${values.sessionPlayerMetaData.distinct_id}
+                        `
+                    }
```

Comment:
here we want to get events with no session id (which implicitly come from not the web sdk so make that explicit)

for the person who was being recorded...

so if we have the person id we use that, otherwise if we have the distinct id we use that

it _should_ be impossible to have neither
like, if we ever have neither so much else is broken that we don't need to guard against it here

Reply:
i shouldn't need to split this query up but the templater really didn't like conditionals here and it worked this way so 🤷